src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -242,45 +242,94 @@ Namespace Microsoft.CodeAnalysis.Semantics
+                Dim initializer As IVariableInitializer = Nothing
+                If first.Kind = BoundKind.LocalDeclaration Then
+                    singleDeclarations = declarationGroup.Cast(Of BoundLocalDeclaration).SelectAsArray(AddressOf GetSingleVariableDeclaration)
+
+                    ' The initializer we use for this group is the initializer attached to the last declaration in this declarator, as that's
+                    ' where it will be parsed in an error case.
+                    ' Initializer is only created if it's not the array initializer for the variable. That initializer is the initializer
+                    ' of the SingleVariableDeclaration child.
+                    Dim last = DirectCast(declarationGroup.Last(), BoundLocalDeclaration)
+                    If last.InitializerOpt IsNot Nothing AndAlso last.InitializerOpt IsNot last.ArrayCreationOpt Then
+                        Debug.Assert(TypeOf last.Syntax Is ModifiedIdentifierSyntax)
+                        Dim initializerValue As IOperation = Create(last.InitializerOpt)
+                        Dim declaratorSyntax = DirectCast(last.Syntax.Parent, VariableDeclaratorSyntax)
+                        Dim initializerSyntax As SyntaxNode = declaratorSyntax.Initializer
+
+                        ' As New clauses with a single variable are bound as BoundLocalDeclarations, so adjust appropriately
                         Dim isImplicit As Boolean = False
-                        If initializerSyntax Is Nothing Then
+                        If initializerSyntax Is Nothing AndAlso
+                           declaratorSyntax.AsClause IsNot Nothing AndAlso
+                           declaratorSyntax.AsClause.IsKind(SyntaxKind.AsNewClause) Then
+                            initializerSyntax = declaratorSyntax.AsClause
+                        ElseIf initializerSyntax Is Nothing Then
                             ' There is no explicit syntax for the initializer, so we use the initializerValue's syntax and mark the operation as implicit.
                             initializerSyntax = initializerValue.Syntax
                             isImplicit = True
                         End If
                         initializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit)
                     End If
-                    builder.Add(OperationFactory.CreateVariableDeclaration(declaration.LocalSymbol, initializer, _semanticModel, declaration.Syntax))
-                ElseIf base.Kind = BoundKind.AsNewLocalDeclarations Then
-                    Dim asNewDeclarations = DirectCast(base, BoundAsNewLocalDeclarations)
-                    Dim localSymbols = asNewDeclarations.LocalDeclarations.SelectAsArray(Of ILocalSymbol)(Function(declaration) declaration.LocalSymbol)
+                ElseIf first.Kind = BoundKind.AsNewLocalDeclarations Then
```

Comment:
>If first.Kind = BoundKind.AsNewLocalDeclarations Then [](start = 20, length = 53)

Either remove the condition, or add an ```Else``` with throw unreachable.

Reply:
Done.

---
In reply to: [146967501](https://github.com/dotnet/roslyn/pull/22842#discussion_r146967501) [](ancestors = 146967501)